integration-tests/jpa-postgresql-withxml/src/test/java/io/quarkus/it/jpa/postgresql/JPAFunctionalityInGraalITCase.java

```diff
@@ -4,30 +4,49 @@
 import org.junit.jupiter.api.Test;
 
 import io.quarkus.test.junit.QuarkusIntegrationTest;
-import io.quarkus.test.junit.nativeimage.ClassInclusionReport;
+import io.quarkus.test.junit.nativeimage.NativeImageReport;
 
 /**
  * Test various JPA operations running in native mode
  */
 @QuarkusIntegrationTest
 public class JPAFunctionalityInGraalITCase extends JPAFunctionalityTest {
 
+    private static final int EXPECTED_NUMBER_OF_USED_CLASSES = 13000;
+    private static final int EXPECTED_NUMBER_OF_USED_METHODS = 85400;
+    private static final int EXPECTED_NUMBER_OF_USED_PACKAGES = 860;
+    private static final int THRESHOLD_PERCENTAGE = 3;
+
     @Test
     public void verifyJDKXMLParsersAreIncluded() {
-        final ClassInclusionReport report = ClassInclusionReport.load();
+        final NativeImageReport usedClassesReport = NativeImageReport.load(NativeImageReport.ReportType.USED_CLASSES);
         //The following classes should be included in this applications;
         //if not, that would be a sign that this test has become too weak
         //to identify the well working of the exclusions.
-        report.assertContains(org.postgresql.jdbc.PgSQLXML.class);
-        report.assertContains(UUIDJdbcType.class);
+        usedClassesReport.assertContains(org.postgresql.jdbc.PgSQLXML.class);
+        usedClassesReport.assertContains(UUIDJdbcType.class);
 
         // And finally verify we included "com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl" which is
         // the fallback implementation class name used in javax.xml.transform.TransformerFactory.newInstance()
         // whose invocation gets triggered when io.quarkus.jdbc.postgresql.runtime.graal.SQLXLMFeature is enabled.
         // We cannot use class javax.xml.transform.TransformerFactory directly since delegation to
         // the implementation might get inlined, thus resulting in 'javax.xml.transform.TransformerFactory'
         // not showing up as a used class in the reports (due to '-H:+InlineBeforeAnalysis').
-        report.assertContains("com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl");
+        usedClassesReport.assertContains("com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl");
+
+        usedClassesReport.assertSizeWithingThreshold(EXPECTED_NUMBER_OF_USED_CLASSES, THRESHOLD_PERCENTAGE);
```

Comment:
nitpicking.. could this assert be moved in a different test?
I wouldn't want the size based assertion to fail one day and give a misleading error that the ` verifyJDKXMLParsersAreIncluded` test failed.

Reply:
Sure, I just wanted to avoid loading the report twice.

Reply:
ah right - initialize the report in a field?